### PR TITLE
ci: add pr-rebase.yaml

### PR DESCRIPTION
**Pull request by kylesomers**
**Issue number:**

### Changes
- Adds a new GitHub workflow that rebases PR branch onto main when triggered manually.
- The workflow uses [a pre-built 3rd party action](https://github.com/marketplace/actions/automatic-rebase).

- When a rebase is needed, add PR comment `/rebase` which will trigger workflow to rebase the current branch onto main.
- Ability to trigger workflow is restricted to members of the GitHub Org. 

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [X] I have performed a self-review of this change
* [ ] Changes have been tested
* [ ] Changes are documented

<details>
<summary>Is this a breaking change?</summary>

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created might not be successful.

### CODE REVIEW

File: .github/workflows/pr-rebase.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Automatic Rebase
```
[the remark this answers is not in the document]
Comment (henrykie) [reply .github/workflows/pr-rebase.yml:1]: @kylesomers should we adjust this or suppress it?

**Review by jorisdon — DISMISSED** [review]
Workflow LGTM, but check the checkov reports and make decision as needed

**Comment by henrykie** [comment]
dependabot has a cool feature where I can `@dependabot rebase` and it'll rebase a branch. I wonder if we could build the same thing with our cloud-game-development-toolkit bot?

**Comment by henrykie** [comment]
https://github.com/chrisledet/rebasebot#setup like this one which just runs on an EC2 instance...

**Review by henrykie — APPROVED** [review]

**Comment by kylesomers** [comment]
@dependabot rebase

**Pull request closed without merging**